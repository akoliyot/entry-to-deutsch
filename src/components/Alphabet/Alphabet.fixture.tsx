import React from 'react';
import { Alphabet } from './Alphabet';
import { useValue } from 'react-cosmos/fixture';

export default () => {
  const [char] = useValue('char', { defaultValue: 'a' });
  return <Alphabet char={char} />;
};
