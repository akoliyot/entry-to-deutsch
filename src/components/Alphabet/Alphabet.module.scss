.root {
  height: 50px;
  width: 50px;
  font-family: 'Fira Code', 'Papyrus';
  display: flex;
  align-items: center;
  justify-content: center;
  border: 1px dashed #dfe833;
  cursor: pointer;
  margin-right: 15px;
  margin-bottom: 15px;
  background-color: transparent;
  transition: background-color 300ms ease-in-out;

  &.active {
    background-color: #f5ff2463;
    // background-color: tomato;
  }

  > p {
    font-size: 42px;
    margin: 0;
    pointer-events: none;
  }

  @keyframes highlight {
    0% {
      background-color: white;
    }

    50% {
      background-color: #cacaca;
    }

    100% {
      background-color: white;
    }
  }
}
