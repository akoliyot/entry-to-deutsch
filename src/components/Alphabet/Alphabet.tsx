import React from 'react';
import styled from 'styled-components';

const StyledAlphabet = styled.div`
  height: 140px;
  width: 140px;
  background-color: crimson;
  font-family: 'Fira Code', 'Papyrus';
`;

interface Props {
  char: string;
}

export const Alphabet: React.FC<Props> = ({ char }) => {
  return (
    <StyledAlphabet>
      <p>StyledAlphabet</p>
    </StyledAlphabet>
  );
};
