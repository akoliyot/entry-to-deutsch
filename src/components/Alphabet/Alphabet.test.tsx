import { Alphabet } from "./Alphabet"
import React from 'react';

describe('Alphabet', () => {
  it('should render correctly', () => {
    expect(<Alphabet isActive={false} char="a" />).toMatchSnapshot();
  })
})