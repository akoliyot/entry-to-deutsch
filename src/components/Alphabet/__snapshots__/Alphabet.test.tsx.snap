// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alphabet should render correctly 1`] = `
<Alphabet
  char="a"
  isActive={false}
/>
`;
